kotest-assertions/src/jvmMain/kotlin/io/kotest/matchers/equality/reflection.kt

```diff
@@ -190,14 +184,10 @@ fun <T : Any> beEqualToIgnoringFields(
    override fun test(value: T): MatcherResult {
 
       val fieldNames = fields.map { it.name }
-      val failed = value::class.memberProperties.filterNot { fieldNames.contains(it.name) }.mapNotNull {
-         val actual = it.getter.call(value)
-         val expected = it.getter.call(other)
-         if (actual == expected) null else {
-            "${it.name}: $actual != $expected"
-         }
-      }
+      val fieldsToBeConsidered: List<KProperty<*>> = value::class.memberProperties.filterNot { fieldNames.contains(it.name) }
+      fieldsToBeConsidered.forEach { it.isAccessible = true }
```

Comment:
Rather than filter it out, if you ask for a field and its private, we should throw an error ?

Reply:
Or simply ignore it in the comparison? I was just asking myself the same.

Reply:
@sksamuel Here user is not asking for including private field directly, user is only saying to ignore some fields(public) which by default resulting in comparing the private fields if there are any.

Reply:
> Or simply ignore it in the comparison? I was just asking myself the same.

But it can very well happen the private field is state maintain by the object which determine its equality.
For example:

```kotlin
class Counter {
 private var currentCount = 0
fun increment(): Unit = currentCount++
fun getCurrentCount():Int = currentCount 
}
```

Reply:
I think if you
* ask for a private field we error
* you ask for public fields but there's also a private field we error

The path should be of least surprise to the user. Silently dropping private fields is probably fine 99% of the time but we should err on the side of caution imo.

Reply:
Another thought in my mind is to let user decide to include or ignore the private fields by using taking a boolean parameter ```ignorePrivateFields``` with default value as false.

Reply:
Yep that would work.

Reply:
Since the signature contain the vararg
```fun <T : Any> T.shouldBeEqualToIgnoringFields(other: T, vararg properties: KProperty<*>)```
So we can't have ```ignorePrivateFields``` as the last parameter with default value. And if put that before the vararg argument then every time we have to  user has to specify that argument. Which will break the existing user code as well that will not look clean.

So going by your suggestion I will make the change to ignore the private fields